### PR TITLE
Add HttpResponse APIs (ClearHeaders/WriteFile/TransmitFile/etc)

### PR DESCRIPTION
**Pull request by twsouthwick**
This change adds a few APIs that were missing from HttpResponse.

SubStatusCode is a placeholder for the value, but does not actually publish it anywhere.

TransmitFile/WriteFile: These APIs are essentially the same, except that they had different memory characteristics. TransmitFile did not buffer, while WriteFile did. On ASP.NET Core, these will just delegate to IHttpResponseBodyFeature.SendFileAsync.

NOTE: This implementation will call Task.GetAwaiter() which should be avoided. We could potentially expose an async API for this on the ASP.NET Core side as a stepping stone if we want.

Part of #91

### CODE REVIEW

File: src/Microsoft.AspNetCore.SystemWebAdapters/HttpResponse.cs [hunk 1]
```diff
@@ -45,14 +45,29 @@ public int StatusCode
             set => _response.StatusCode = value;
         }
 
-        public string StatusDescription
+        public int SubStatusCode { get; set; } = -1;
```
Comment (mjrousos) [inline src/Microsoft.AspNetCore.SystemWebAdapters/HttpResponse.cs:48]: Why -1? It looks like this defaulted to 0 in ASP.NET

**Review by mjrousos — CHANGES_REQUESTED** [review]
Looks good other than a question about SubStatusCode's initial value

File: src/Microsoft.AspNetCore.SystemWebAdapters/HttpResponse.cs (hunk 1 above)
Comment (twsouthwick) [reply src/Microsoft.AspNetCore.SystemWebAdapters/HttpResponse.cs]: good catch. there was a different value that had a default as -1 and I think I got that mixed up

**Review by mjrousos — APPROVED** [review]
LGTM

File: src/Microsoft.AspNetCore.SystemWebAdapters/HttpResponse.cs [hunk 2]
```diff
@@ -167,6 +182,26 @@ public void AppendHeader(string name, string value)
             }
         }
 
+        public bool IsRequestBeingRedirected { get; private set; }
```
[the remark this answers is not in the document]
Comment (twsouthwick) [reply src/Microsoft.AspNetCore.SystemWebAdapters/HttpResponse.cs]: done

**Pull request merged**